### PR TITLE
Dashboard set up

### PR DESCRIPTION
**Pull request by nicorithner**
### Description:

  Followed issue requirements to create 'User Dashboard' basic set up
### Tasks:

- [x] Added requirements to the view
  - [x] Discover Movies button
  - [x] Sections titles
- [x] Refactor to dry code based on Rubocop requests
  - [x] Added sessions_helper
  - [x] Refactor create method

#### Notes:
  Used sessions_helper as module to dry up code, however we wonder whether a module that is used only by one controller was appropriate.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,16 +1,6 @@
 class SessionsController < ApplicationController
+  include SessionsHelper
```
Comment (nicorithner) [inline app/controllers/sessions_controller.rb:2]: Used sessions_helper as module to dry up code, however we wonder whether a module that is used only by one controller was appropriate.

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module SessionsHelper
+  def create_session
+    if (user = User.find_by(email: params[:email]))
```
Comment (nicorithner) [inline app/helpers/sessions_helper.rb:3]: wrapped statement in parenthesis following Rubocop suggestion

File: app/models/user.rb [hunk 3]
```diff
@@ -5,4 +5,8 @@ class User < ApplicationRecord
   has_secure_password
 
   enum role: { user: 0 }
+
+  def humanize_username
+    email.split('@').first
```
Comment (nicorithner) [inline app/models/user.rb:10]: Removed redundant `self.`

File: app/views/user/dashboard/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<h1>Welcome <%= current_user.humanize_username %>!</h1>
+
+<%= button_to "Discover Movies", '#' %>
+
+<div class="friends">
```
Comment (nicorithner) [inline app/views/user/dashboard/index.html.erb:5]: Starting placeholder for upcoming Friends feature

File: app/views/user/dashboard/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  <h4>Friends Section</h4>
```
Comment (nicorithner) [inline app/views/user/dashboard/index.html.erb:6]: Starting placeholder for upcoming Parties feature

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (hashmaster3k) [reply app/controllers/sessions_controller.rb:2]: Good question. @iandouglas, thoughts?

**Review by hashmaster3k — APPROVED** [review]

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (iandouglas) [reply app/controllers/sessions_controller.rb:2]: I usually don't pull code out into a module until it's used in more than one place, then I look for ways to DRY up that code. I'd also typically make helper methods that return some sort of status and use that status to let the controller figure out what to do, like having it return if it succeeded or not, and if not, what error, and where to redirect the user, and let the controller actually set the flash message and do the redirection. In this case you have to do some digging to figure out what's going on.

**Pull request merged**